services/src/test/java/io/scalecube/services/ServiceAuthLocalTest.java

```diff
@@ -0,0 +1,186 @@
+package io.scalecube.services;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import io.scalecube.services.auth.Authenticator;
+import io.scalecube.services.exceptions.UnauthorizedException;
+import io.scalecube.services.sut.security.PartiallySecuredService;
+import io.scalecube.services.sut.security.PartiallySecuredServiceImpl;
+import io.scalecube.services.sut.security.SecuredService;
+import io.scalecube.services.sut.security.SecuredServiceImpl;
+import io.scalecube.services.sut.security.UserProfile;
+import java.time.Duration;
+import java.util.function.Consumer;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+final class ServiceAuthLocalTest {
+
+  private static final Duration TIMEOUT = Duration.ofSeconds(3);
+
+  private static final String CREDENTIALS = "valid_credentials";
+  private static final Authenticator<String, UserProfile> authenticator =
+      credentials -> {
+        if (CREDENTIALS.equals(credentials)) {
+          return Mono.just(new UserProfile("Alice", "ADMIN"));
+        }
+
+        return Mono.error(new UnauthorizedException("Authentication failed"));
```

Comment:
`UnauthorizedException`? its authentication, not authorization.

Reply:
`UnathorizedException` is one of default `scalecube-service` exceptions.
Those exceptions have similar semantics to HTTP error codes (also, `errorType` of exception has exactly the same value as HTTP error code).

In HTTP:
- if authentication fails `401 Unauthorized` should be returned (see https://tools.ietf.org/html/rfc7235#section-3.1).
- if authorization fails `403 Forbidden` should be returned (see https://tools.ietf.org/html/rfc7231#section-6.5.3)

If you would like to introduce a new type of exception(s), please discuss it with @artem-v.